[BEAM-975] Improve default connection options, javadoc and style in MongoDbIO

jbonofre opened the pull request:
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [X] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [X] Make sure tests pass via `mvn clean verify`.
 - [X] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [X] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

jbonofre commented:
R: @dhalperi 
CC: @davorbonaci

dhalperi commented on sdks/java/io/mongodb/src/main/java/org/apache/beam/sdk/io/mongodb/MongoDbIO.java:
```diff
@@ -218,61 +298,64 @@ public void populateDisplayData(DisplayData.Builder builder) {
+        if (spec.numSplits() > 0) {
+          // the user defines his desired number of splits
+          // calculate the batch size
+          long estimatedSizeBytes = getEstimatedSizeBytes(options);
```
do you want a variant where you can pass the already-existing client into the connection? That way you don't have to open two connections here, you can reuse the existing one for estimating size.

davorbonaci commented:
Any updates?

jbonofre commented:
I'm updating this pull request.

jbonofre commented:
@dhalperi @davorbonaci I updated this pull request:
1. rebased on master
2. fix javadoc
3. be able to use an existing Mongo client in `estimatedSizeBytes()`

jbonofre commented:
Thanks, let me add a couple of tests.

jbonofre commented:
I'm waiting Jenkins build before merging.
